src/libraries/System.Net.NameResolution/src/System/Net/NameResolutionPal.Windows.cs

```diff
@@ -164,12 +164,25 @@ public static unsafe Task GetAddrInfoAsync(string hostName, bool justAddresses,
             {
                 state.RegisterForCancellation(cancellationToken);
             }
+            else if (errorCode == SocketError.TryAgain)
+            {
+                // WSATRY_AGAIN indicates possible problem with reachability according to docs.
+                // However, if servers are really unreachable, we would still get IOPending here
+                // and final result would be posted via overlapped IO.
```

Comment:
I'm not understanding the above comment.

Reply:
WSATRY_AGAIN is valued error code for failures other than we see in the impersonification case. What I'm trying to say here is based on my testing and understanding, the _other_ WSATRY_AGAIN would not be synchronous failure. 
In such cases, we should return transient failure to caller rather than re-trying on thread-pool IMHO.